### PR TITLE
scx_layered: Implement in-layer execution protection to replace cost based fairness

### PR DESCRIPTION
**Pull request by htejun**
cost based fairness mechanism tries to implement weight-based fairness by accounting the consumption of each layer and changing preemption and dispatch behaviors accordingly on each scheduling event. While this approach can work, it's unnecessarily finely grained given that `scx_layered` already has control over which layers get how many CPUs.

Previous changes made CPU allocations to follow the layer weights under system saturation. This PR implement in-layer execution protection so that layers can reliably grow and fully occupy its CPUs unaffected by preempt layers. Weight-based fairness is enforced through the combination of CPU allocation and in-layer execution protection - a layer is guaranteed to have its fair share number of CPUs according to its weight under contention and that it can use all its CPUs.

While there are a couple missing pieces, the basic mechanism seems to work as expected. Drop cost based fairness mechanism. This reduces code complexity and `.stopping()` overhead and makes future changes easier.

### CODE REVIEW

File: scheds/rust/scx_layered/src/bpf/main.bpf.c [hunk 1]
```diff
@@ -2738,10 +2674,12 @@ SCX_OPS_DEFINE(layered,
...
 	       .set_weight		= (void *)layered_set_weight,
 	       .set_cpumask		= (void *)layered_set_cpumask,
+	       .update_idle		= (void *)layered_update_idle,
 	       .cpu_release		= (void *)layered_cpu_release,
 	       .init_task		= (void *)layered_init_task,
 	       .exit_task		= (void *)layered_exit_task,
 	       .dump			= (void *)layered_dump,
 	       .init			= (void *)layered_init,
 	       .exit			= (void *)layered_exit,
+	       .flags			= SCX_OPS_KEEP_BUILTIN_IDLE,
```
Comment (hodgesds) [inline scheds/rust/scx_layered/src/bpf/main.bpf.c:2684]: Is this flag available for backports as well?

**Review by hodgesds — APPROVED** [review]
I think this makes sense, it's definitely far easier to verify and maintain.

File: scheds/rust/scx_layered/src/bpf/main.bpf.c (hunk 1 above)
Comment (htejun) [reply scheds/rust/scx_layered/src/bpf/main.bpf.c:2684]: Yeah, that flag has been around for a long time.

**Review by likewhatevs — APPROVED** [review]
I like complexity being pushed to rust side and I can follow what this is doing. Also fewer bpf.c and bpf.h files is nice too.

**Comment by likewhatevs** [comment]
I've been looking at this PR merging for what feels like all day so I'm going to just merge it, something weird up w/ bpfland job, rest is good.

**Pull request merged**